user_install.sh

```diff
@@ -0,0 +1,404 @@
+	echo ""
+	case "$RESPONSE" in
+		"1")
+			if [ ! -d $CATKIN_DIR/src/Sub8 ]; then
+				echo -n "User fork URI for the Sub8 repository: " && read SUB_USER_FORK
+			fi
+			INSTALL_SUB=true
+			SELECTED=true
+			echo ""
+		;;
+		"2")
+			echo "The PropaGator project has not been worked on since the dark ages of MIL, so it"
+			echo "is not supported by this script."
+			echo ""
+		;;
+		"3")
+			echo "The NaviGator project was developed on Ubuntu 14.04 with ROS Indigo. Several"
+			echo "dependencies no longer exist in ROS Kinetic, so in order to install it, ROS"
+			echo "Indigo, the Sub8 repository at an earlier date,  and all of the old Sub8"
+			echo "dependencies will need to be downloaded and installed."
+			echo -n "Do you still wish to proceed? [y/N] " && read RESPONSE
+			if ([ "$RESPONSE" = "Y" ] || [ "$RESPONSE" = "y" ]); then
+				if [ ! -d $CATKIN_DIR/src/Navigator ]; then
+					echo -n "User fork URI for the Navigator repository: " && read NAV_USER_FORK
+				fi
+				INSTALL_NAV=true
+				SELECTED=true
+			fi
+			echo ""
+		;;
+		"")
+			echo "You must select one of the projects by entering it's number on the list"
+			echo ""
+		;;
+		*)
+			echo "$RESPONSE is not a valid selection"
+			echo ""
+		;;
+	esac
+done
+
+# Prompt the user to install the BlueView SDK if it is not already installed
+if [ ! -d $CATKIN_DIR/src/bvtsdk ]; then
+	echo "The BlueView SDK used to interface with the Teledyne imaging sonar is encrypted"
+	echo "in order to protect the intellectual property of BlueView. If you will be doing"
+	echo "work with the imaging sonar on your machine, it is recommended that you install"
+	echo "this now. If not, you probably do not need to."
+	echo -n "Do you wish to install the SDK? [y/N] " && read RESPONSE
+	echo ""
+
+	# If the user chooses to install the BlueView SDK, retrieve the password from them
+	if ([ "$RESPONSE" = "Y" ] || [ "$RESPONSE" = "y" ]); then
+		echo "The SDK is encrypted with a password. You need to obtain this password from one"
+		echo "of the senior members of MIL."
+		echo -n "Encryption password: " && read -s BVSDK_PASSWORD
+		echo ""
+		echo ""
+	fi
+fi
+
+
+#==================#
+# Pre-Flight Check #
+#==================#
+
+instlog "Starting the pre-flight system check to ensure installation was done properly"
+
+# Check whether or not github.com is reachable
+# This also makes sure that the user is connected to the internet
+if [ "`check_host github.com`" = "true" ]; then
+	NET_CHECK=true
+	echo -n "[ " && instpass && echo -n "] "
+else
+	NET_CHECK=false
+	echo -n "[ " && instfail && echo -n "] "
+fi
+echo "Internet connectivity check"
+
+if !($NET_CHECK); then
+
+	# The script will not allow the user to install without internet
+	instwarn "Terminating installation due to the lack of an internet connection"
+	instwarn "The install script needs to be able to connect to Github and other sites"
+	exit 1
+fi
+
+# Set the required OS based on inputs and installed distribution
+if ($INSTALL_NAV); then
```

Comment:
I'm wondering if having some kind of config would be a good idea where externally you could specify version dependencies and things like this. It would make it easier to change what versions are supported with which repos (though may add way too much complexity in the actually script here)